### PR TITLE
Add Remain HP option 0% (1HP) - Conjunction-like

### PR DESCRIPTION
**Pull request by h-yasha**
#62 
> バフ：残りHP割合 に「強制HP1」

<hr>

- [x] Add option for 0% (1HP).
    - [ ] add description for `0` to note it is limited to be at least 1HP.
- [x] Limit HP to be 1 when remainHP is set to 0%.
    - in "usual" cases this would be unnecessary? the number would be rather small.  _1 / 20000 = 0.00005_, and calculating with remainHP set to **0** does not break anything from my tests. While it's still "less" accurate, most results are rounded (in display at least). and could refrain from changing (moving) this much code.

### CODE REVIEW

**Comment by OrkunKocyigit** [comment]
Do you think you can add 0% point to chart and damage simulator as well? I am assuming chart is our userbase biggest use case for emnity calculation and that would be helpful

**Comment by h-yasha** [comment]
> Do you think you can add 0% point to chart and damage simulator as well? I am assuming chart is our userbase biggest use case for emnity calculation and that would be helpful

- [ ] add 0% to enmity graph.
- [ ] add 0% to remainHP in damage simulator (I expected it to be the same.. apparently not)

**Comment by OrkunKocyigit** [comment]
I'll try to review this change.
- [ ] Check With 0 Emnity Skill
- [ ] Check With Emnity Skill
- [ ] Chart
- [ ] Simulator
- [ ] Code Style

**Comment by OrkunKocyigit** [comment]
@h-yasha Simulator crashes.
Uncaught TypeError: Cannot read property '0' of undefined
    at Object.makeTurnBuff (simulator.js:270)
    at Object.render (simulator.js:663)
    at finishClassComponent (react-dom.development.js:14301)
    at updateClassComponent (react-dom.development.js:14264)
    at beginWork (react-dom.development.js:15082)
    at performUnitOfWork (react-dom.development.js:17820)
    at workLoop (react-dom.development.js:17860)
    at HTMLUnknownElement.callCallback (react-dom.development.js:149)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:199)
    at invokeGuardedCallback (react-dom.development.js:256)

**Comment by OrkunKocyigit** [comment]
@h-yasha 
There is a inconsistency between chart data and results.js.
Results.js
image

Chart.js
image

**Comment by h-yasha** [comment]
> @h-yasha
> There is a inconsistency between chart data and results.js.
> Results.js
> image
> 
> Chart.js
> image

in enmity graph it expects _everyone_ to be at that HP%. so having a character be at different HP% would result into different results

**Comment by h-yasha** [comment]
> @h-yasha Simulator crashes.
> Uncaught TypeError: Cannot read property '0' of undefined
> at Object.makeTurnBuff (simulator.js:270)
> at Object.render (simulator.js:663)
> at finishClassComponent (react-dom.development.js:14301)
> at updateClassComponent (react-dom.development.js:14264)
> at beginWork (react-dom.development.js:15082)
> at performUnitOfWork (react-dom.development.js:17820)
> at workLoop (react-dom.development.js:17860)
> at HTMLUnknownElement.callCallback (react-dom.development.js:149)
> at Object.invokeGuardedCallbackDev (react-dom.development.js:199)
> at invokeGuardedCallback (react-dom.development.js:256)
> 
> Steps to reproduce:
> 
> 1. Set any characters hp percentage to 0%
> 2. Click on simulator

after trying to reproduce, 
- This bug is related to having multiple characters with the same name rather than remainHP being 0%.

**Comment by OrkunKocyigit** [comment]
Oops sorry, I'll retest without using same characters

File: src/global_logic.js [hunk 1]
```diff
@@ -2793,14 +2797,14 @@ module.exports.generateHaisuiData = function (res, arml, summon, prof, chara, st
 
     // Because the character formation is unchanged every weapon organization, it can be calculated earlier
     var charaHaisuiBuff = [];
-    for (var k = 0; k < 100; ++k) {
-        let charaHaisuiValue = module.exports.recalcCharaHaisui(chara, 0.01 * (k + 1));
+    for (let k = 0; k <= 100; k++) {
+        let charaHaisuiValue = module.exports.recalcCharaHaisui(chara, 0.01 * (k));
```
Comment (OrkunKocyigit) [inline src/global_logic.js]: Redundant parentheses

File: src/global_logic.js [hunk 2]
```diff
@@ -2793,14 +2797,14 @@ module.exports.generateHaisuiData = function (res, arml, summon, prof, chara, st
...
         charaHaisuiBuff.push(charaHaisuiValue);
     }
 
     var normalSupportKonshin = [];
-    for (var k = 0; k < 100; ++k) {
-        let normalSupportKonshinValue = module.exports.recalcNormalSupportKonshin(chara, 0.01 * (k + 1));
+    for (let k = 0; k <= 100; k++) {
+        let normalSupportKonshinValue = module.exports.recalcNormalSupportKonshin(chara, 0.01 * (k));
```
Comment (OrkunKocyigit) [inline src/global_logic.js]: Redundant parentheses

File: src/global_logic.js [hunk 3]
```diff
@@ -2862,8 +2866,8 @@ module.exports.generateHaisuiData = function (res, arml, summon, prof, chara, st
 
                 var lbHaisuiBuff = [],
                     lbKonshinBuff = [];
-                for (let k = 0; k < 100; ++k) {
-                    let hp = 0.01 * (k + 1);
+                for (let k = 0; k <= 100; k++) {
+                    let hp = 0.01 * (k);
```
Comment (OrkunKocyigit) [inline src/global_logic.js]: Redundant parentheses

**Comment by OrkunKocyigit** [comment]
@h-yasha BTW do you plan to add "add description for 0 to note it is limited to be at least 1HP.", I really don't think it is required as 0% used by both wiki and gamehive but I'll wait before I'll approve if you want to add that.

**Comment by h-yasha** [comment]
> @h-yasha BTW do you plan to add "add description for 0 to note it is limited to be at least 1HP.", I really don't think it is required as 0% used by both wiki and gamehive but I'll wait before I'll approve if you want to add that.

@OrkunKocyigit I do not think this is necessary, I (others too, hopefully) don't expect 0% remaining HP to be actually **0** HP.

**Pull request merged**